### PR TITLE
Add hidden fields to change details and cancel interview links

### PR DESCRIPTION
**Pull request by richardpattinson**
## Context

Add hidden fields to the change details/ cancel interviews link in the interview course and summary component. The aim of this change is to improve accessibility for visually impaired users with screen readers. 

None.

Run locally and inspect element on the links, they should have a hidden field containing the date and time of the interview. 

## Link to Trello Card

https://trello.com/c/1Ul8zg0Z/3411-add-hidden-text-to-interview-summary-cards

## Things to check

- [X] This code does not rely on migrations in the same Pull Request
- [X] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [X] API release notes have been updated if necessary
- [X] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: spec/components/provider_interface/interview_and_course_summary_component_spec.rb [hunk 1]
```diff
@@ -24,6 +24,16 @@
     expect(component).to include(interview.location)
   end
 
+  it 'applies a hidden field to the change link' do
```
Comment (despo) [inline spec/components/provider_interface/interview_and_course_summary_component_spec.rb:27]: If we are to test this our spec should indicate why this is important.

```suggestion
  it 'change link has a hidden field to improve accessibility for visually impaired users' do
```

File: spec/components/provider_interface/interview_and_course_summary_component_spec.rb [hunk 2]
```diff
@@ -24,6 +24,16 @@
...
+    component = render_inline(described_class.new(interview: interview, user_can_change_interview: true))
+    expect(component.css('.govuk-visually-hidden').first.text).to include(interview.date_and_time.to_s(:govuk_date_and_time))
+  end
+
+  it 'applies a hidden field to the cancel link' do
```
Comment (despo) [inline spec/components/provider_interface/interview_and_course_summary_component_spec.rb:32]: If we are to test this our spec should indicate why this is important.

```suggestion
  it 'cancel link has a hidden field to improve accessibility for visually impaired users' do
```

**Review by despo — APPROVED** [review]
The spec descriptions can be improved but other than that it looks good :shipit:

File: app/components/provider_interface/interview_and_course_summary_component.html.erb [hunk 3]
```diff
@@ -5,10 +5,16 @@
         <div class="app-summary-card__actions">
           <ul class="app-summary-card__actions-list">
             <li class="app-summary-card__actions-list-item">
-              <%= govuk_link_to 'Change details', edit_provider_interface_application_choice_interview_path(application_choice, interview), class: 'app-summary-card__actions-list-item govuk-link--no-visited-state' %>
+              <%= link_to edit_provider_interface_application_choice_interview_path(application_choice, interview), class: 'app-summary-card__actions-list-item govuk-link govuk-link--no-visited-state' do %>
+                <span class="govuk-visually-hidden"> <%= interview.date_and_time.to_s(:govuk_date_and_time) %> </span>
```
Comment (duncanjbrown) [inline app/components/provider_interface/interview_and_course_summary_component.html.erb:9]:
```suggestion
                <span class="govuk-visually-hidden"><%= interview.date_and_time.to_s(:govuk_date_and_time) %> </span>
```

File: app/components/provider_interface/interview_and_course_summary_component.html.erb [hunk 4]
```diff
@@ -5,10 +5,16 @@
...
+                Change details
+              <% end %>
             </li>
             <li class="app-summary-card__actions-list-item">
-              <%= govuk_link_to 'Cancel interview', cancel_provider_interface_application_choice_interview_path(application_choice, interview), class: 'app-summary-card__actions-list-item govuk-link--no-visited-state' %>
+              <%= link_to cancel_provider_interface_application_choice_interview_path(application_choice, interview), class: 'app-summary-card__actions-list-item govuk-link govuk-link--no-visited-state' do %>
+                <span class="govuk-visually-hidden"> <%= interview.date_and_time.to_s(:govuk_date_and_time) %> </span>
```
Comment (duncanjbrown) [inline app/components/provider_interface/interview_and_course_summary_component.html.erb:15]:
```suggestion
                <span class="govuk-visually-hidden"><%= interview.date_and_time.to_s(:govuk_date_and_time) %> </span>
```

**Review by duncanjbrown — APPROVED** [review]
Looks good! Could remove a couple of spaces 📐

File: app/components/provider_interface/interview_and_course_summary_component.html.erb [hunk 5]
```diff
@@ -5,10 +5,16 @@
         <div class="app-summary-card__actions">
           <ul class="app-summary-card__actions-list">
             <li class="app-summary-card__actions-list-item">
-              <%= govuk_link_to 'Change details', edit_provider_interface_application_choice_interview_path(application_choice, interview), class: 'app-summary-card__actions-list-item govuk-link--no-visited-state' %>
+              <%= link_to edit_provider_interface_application_choice_interview_path(application_choice, interview), class: 'app-summary-card__actions-list-item govuk-link govuk-link--no-visited-state' do %>
```
Comment (despo) [inline app/components/provider_interface/interview_and_course_summary_component.html.erb:8]: I missed this in my previous review. Is there any reason we are no longer using `govuk_link_to` here?

File: app/components/provider_interface/interview_and_course_summary_component.html.erb [hunk 6]
```diff
@@ -5,10 +5,16 @@
...
+                <span class="govuk-visually-hidden"> <%= interview.date_and_time.to_s(:govuk_date_and_time) %> </span>
+                Change details
+              <% end %>
             </li>
             <li class="app-summary-card__actions-list-item">
-              <%= govuk_link_to 'Cancel interview', cancel_provider_interface_application_choice_interview_path(application_choice, interview), class: 'app-summary-card__actions-list-item govuk-link--no-visited-state' %>
+              <%= link_to cancel_provider_interface_application_choice_interview_path(application_choice, interview), class: 'app-summary-card__actions-list-item govuk-link govuk-link--no-visited-state' do %>
+                <span class="govuk-visually-hidden"> <%= interview.date_and_time.to_s(:govuk_date_and_time) %> </span>
+                Cancel interview
```
Comment (AgaDufrat) [inline app/components/provider_interface/interview_and_course_summary_component.html.erb:16]: In the prototype the hidden span is placed after the visible link text. We use this pattern when making all links unique.
Screenshot 2021-02-23 at 12 34 10

**Pull request merged**